src/pattern-library/components/patterns/DialogComponents.js

```diff
@@ -66,6 +74,19 @@ const showDialog = ({ DialogComponent, container, setOpen, props }) => {
 };
 
 export default function DialogComponents() {
+  // Dialog/Modal state for each of the examples
+
+  // Basic Dialog example
+  const [dialogIsOpen, setDialogIsOpen] = useState(false);
+  // Basic Modal example
+  const [, setModalIsOpen] = useState(false);
+  // Modal with manual focus control
+  const [, setFocusModalIsOpen] = useState(false);
+  // Modal with overflowing content example
+  const [, setLongModalIsOpen] = useState(false);
+  // ConfirmModal example
+  const [, setConfirmModalIsOpen] = useState(false);
```

Comment:
What is the rationale for these write-only state values? Is it just to provide a dummy callback to pass to `showDialog`'s `setOpen` prop? If so you could make that optional instead.

Reply:
I mentioned [in another comment](https://github.com/hypothesis/frontend-shared/pull/146#discussion_r670539602) that this needs some attention (i.e. these demos need refactor). I'd like to tackle this separately so I don't get too mired.

Reply:
OK, but I didn't grok the thinking behind this code. The only scenario in which write-only state is useful that I can see is if you want to force a re-render even though nothing the component directly depends on changed.